src/cider/type/plan/FunctionExpr.cpp

```diff
@@ -0,0 +1,307 @@
+        (func_ti.is_bytes() && arg_ti.is_bytes())) {
+      // If the function returns an array and any of the arguments are arrays, allow NULL
+      // scalars.
+      // TODO: Make this a property of the FunctionOper following `RETURN NULL ON NULL`
+      // semantics.
+      return false;
+    } else if (!arg_ti.get_notnull() && !arg_ti.is_buffer()) {
+      return true;
+    } else {
+      continue;
+    }
+  }
+  return false;
+}
+
+inline SQLTypeInfo get_sql_type_from_JIT_type_tag(const JITTypeTag jit_type) {
+  switch (jit_type) {
+    case JITTypeTag::FLOAT:
+      return SQLTypeInfo(kFLOAT, false);
+    case JITTypeTag::DOUBLE:
+      return SQLTypeInfo(kDOUBLE, false);
+    case JITTypeTag::BOOL:
+      return SQLTypeInfo(kBOOLEAN, false);
+    case JITTypeTag::INT8:
+      return SQLTypeInfo(kTINYINT, false);
+    case JITTypeTag::INT16:
+      return SQLTypeInfo(kSMALLINT, false);
+    case JITTypeTag::INT32:
+      return SQLTypeInfo(kINT, false);
+    case JITTypeTag::INT64:
+      return SQLTypeInfo(kBIGINT, false);
+    default:
+      LOG(FATAL) << "Unrecognized jit type for SQL type";
+  }
+  UNREACHABLE();
+  return SQLTypeInfo();
+}
+
+// Generates code which returns true if at least one of the arguments is NULL.
+JITValue* codegenFunctionOperNullArgForArrow(
+    JITFunction& func,
+    const Analyzer::FunctionOper* function_oper,
+    const std::vector<JITValue*>& orig_arg_lv_nulls) {
+  auto one_arg_null = func.createVariable(JITTypeTag::BOOL, "one_arg_null", false);
+  size_t physical_coord_cols = 0;
+  for (size_t i = 0, j = 0; i < function_oper->getArity();
+       ++i, j += std::max(size_t(1), physical_coord_cols)) {
+    const auto arg = function_oper->getArg(i);
+    const auto& arg_ti = arg->get_type_info();
+    physical_coord_cols = arg_ti.get_physical_coord_cols();
+    if (arg_ti.get_notnull()) {
+      continue;
+    }
+    // TODO:(yma11) add string support
+    CHECK(arg_ti.is_number() or arg_ti.is_boolean());
+    one_arg_null = one_arg_null->orOp(*orig_arg_lv_nulls[j]);
+  }
+  return one_arg_null.get();
+}
+
+const Analyzer::Expr* expr_rewrite_to_cast(const Analyzer::Expr* orig_expr,
+                                           SQLTypes target_type) {
+  std::shared_ptr<Analyzer::Expr> ptr(const_cast<Analyzer::Expr*>(orig_expr));
+  return std::make_shared<Analyzer::UOper>(SQLTypeInfo(target_type), false, kCAST, ptr)
+      .get();
+}
+
+JITExprValue& FunctionOper::codegen(CodegenContext& context) {
+  JITFunction& func = *context.getJITFunction();
+  ExtensionFunction ext_func_sig = [=]() {
+    try {
+      return bind_function(this);
+    } catch (CiderCompileException& e) {
+      LOG(WARNING) << "codegenFunctionOper[CPU]: " << e.what();
+      throw;
+    }
+  }();
+  const auto& ret_ti = this->get_type_info();
+  CHECK(ret_ti.is_integer() || ret_ti.is_fp() || ret_ti.is_boolean());
+  auto ret_ty = ext_arg_type_to_JIT_type_tag(ext_func_sig.getRet());
```

Comment:
I think you don't need to compare type info through `JIT_type_tag`. Do the compare based on info of the `FuncitionOper` itself.